# issue 20630

事情的缘起是 issue 20630。遇到的 bug 是，如果用户中止 http2 连接， http2 连接的 `close(cb)` 回调不会被调用。

```js
'use strict';

const http2 = require('http2');
const assert = require('assert');

// 创建 http2 服务器
const server = http2.createServer();
let client;

// port=0时 server 随意分配端口
server.listen(0, function() {
  // 创建 http2 客户端
  client = http2.connect(`http://localhost:${server.address().port}`);
  // 注册 on connect 回调
  client.on('connect', function() {
    console.log('connect');

    server.close(function() {
      console.log('the close callback');
    });
  });
});

server.on('session', (s) => {
  s.setTimeout(10, function () {
    console.log('timeout');
    s.destroy();
  });
});
```

